pkg/compiler/consts.go

```diff
@@ -123,6 +123,9 @@ func (comp *compiler) addConst(infos map[string]*constInfo, pos ast.Pos, name st
 	if _, builtin := comp.builtinConsts[name]; builtin {
 		return
 	}
+	if _, isFlag := comp.intFlags[name]; isFlag {
```

Comment:
The constant extraction logic goes over all type arguments of kind `kindInt` (function `extractTypeConsts` above) and considers any that is an identifier (`.Ident != ""`) has a constant.

Unfortunately, flags can now be used in int types, via an argument of kind `kindInt | kindIdent`, which will match the above conditions. So to avoid flags defined via `int` from being considered constants, we must skip them here.

For example, removing those three lines causes the following errors:
```
bin/syz-sysgen
compilation of linux/386 target failed:
sys/linux/bpf.txt:478:34: bpf_alu_insn is defined for none of the arches
sys/linux/bpf.txt:478:64: bpf_alu_op is defined for none of the arches
sys/linux/bpf.txt:478:48: bpf_alu_source is defined for none of the arches
sys/linux/bpf.txt:449:36: bpf_ldst_insn is defined for none of the arches
sys/linux/bpf.txt:449:66: bpf_ldst_mode is defined for none of the arches
sys/linux/bpf.txt:449:51: bpf_ldst_size is defined for none of the arches
```

Reply:
I see, thanks!

Reply:
I'll add a comment before the check to clarify this to future readers.

Reply:
Done. I repushed with the comment, without rebasing.